extras/indexing/src/main/java/mvm/rya/sail/config/RyaSailFactory.java

```diff
@@ -55,91 +52,98 @@
         return getRyaSail(conf);
     }
 
-    private static Sail getRyaSail(final Configuration config) throws AccumuloException, AccumuloSecurityException, RyaDAOException, InferenceEngineException {
+    private static Sail getRyaSail(final Configuration config) throws InferenceEngineException, RyaDAOException, AccumuloException, AccumuloSecurityException {
         final RdfCloudTripleStore store = new RdfCloudTripleStore();
-        RyaDAO crdfdao = null;
-        RdfCloudTripleStoreConfiguration conf;
+        final RyaDAO dao;
+        final RdfCloudTripleStoreConfiguration conf;
 
-        RyaDetailsRepository ryaDetailsRepo;
         final String user = config.get(ConfigUtils.CLOUDBASE_USER);
         final String pswd = config.get(ConfigUtils.CLOUDBASE_PASSWORD);
         final String instance = config.get(ConfigUtils.CLOUDBASE_INSTANCE);
-        if (ConfigUtils.getUseMongo(config)) {
+        if(ConfigUtils.getUseMongo(config)) {
             conf = new MongoDBRdfConfiguration(config);
-            conf.setTablePrefix(config.get(RdfCloudTripleStoreConfiguration.CONF_TBL_PREFIX));
-            final MongoDBRdfConfiguration mongoConf = (MongoDBRdfConfiguration) conf;
-            final MongoCredential creds =
-                    MongoCredential.createCredential(user, instance, pswd.toCharArray());
-            final String hostname = ((MongoDBRdfConfiguration)conf).getMongoInstance();
-            final int port = Integer.parseInt(((MongoDBRdfConfiguration)conf).getMongoPort());
-
-            MongoClient client = null;
-            try {
-                client = new MongoClient(
-                    new ServerAddress(hostname, port),
-                    Lists.newArrayList(creds));
-                ryaDetailsRepo = new MongoRyaInstanceDetailsRepository(client, mongoConf.getCollectionName());
-                RyaDetailsToConfiguration.addRyaDetailsToConfiguration(ryaDetailsRepo.getRyaInstanceDetails(), mongoConf);
-            } catch(final RyaDetailsRepositoryException e) {
-                Log.info("Instance does not have a rya details collection, skipping.");
-            } catch (final UnknownHostException ue) {
-                throw new RyaDAOException("Unable to connect to mongo at the configured location.", ue);
-            } finally {
-                ConfigUtils.setIndexers(conf);
-
-                if(client != null) {
-                    crdfdao = new MongoDBRyaDAO(mongoConf, client);
-                } else {
-                    try {
-                        crdfdao = new MongoDBRyaDAO(mongoConf);
-                    } catch (NumberFormatException | UnknownHostException e) {
-                        e.printStackTrace();
-                    }
-                }
-                crdfdao.init();
-
-                conf.setDisplayQueryPlan(true);
-                store.setRyaDAO(crdfdao);
-            }
+            final MongoClient client = updateMongoConfig((MongoDBRdfConfiguration) conf, user, pswd, instance);
```

Comment:
Yes. The Accumulo Instance Name is the the name that IDs Accumulo within Zookeeper. The Rya Instance Name is the name of the Rya instance hosted by Accumulo.